### PR TITLE
Add support for MERGED_WITH_CASH_TRANSACTION system message

### PR DESCRIPTION
**Pull request by yuwenmemon**
cc @arosiclair 

HOLD ON https://github.com/Expensify/Auth/pull/10726

### Details
Screenshot 2024-05-01 at 9 12 21 PM

Add support for the MERGED_WITH_CASH_TRANSACTION system message in NewDot.

### Fixed Issues
$ https://github.com/Expensify/App/issues/39540

### Tests

1. Follow the steps in [this SO](https://stackoverflowteams.com/c/expensify/questions/194) to create a merged transaction with SmartScan / Card Import
2. Ensure that once the card is scraped, and the transaction is merged, a comment appears on the transaction thread of the report indicating that it had been matched with a card transaction:
Screenshot 2024-05-01 at 9 12 21 PM

### Offline tests
- None

### QA Steps
- Ping @yuwenmemon to test
---

1. Add a personal card to start importing transactions into Expensify if you have not done so already. (https://help.expensify.com/articles/expensify-classic/connect-credit-cards/Personal-Credit-Cards#:~:text=Log%20into%20your%20Expensify%20account,bank%20from%20the%20list%20provided.)
2. Have a policy that is workspace chat enabled.
3. SmartScan a receipt for a purchase made with that same personal credit card. Make sure you do this **before** running an import of your personal card. Add that smartscanned expense to your workspace chat enabled policy via a report.
4. Once the receipt is smartscanned, you should see it has the 7-day hold violation ("pending match with card transaction")
5. Run an import of transactions from your card. You may have to wait several days for the transaction to post.
6. Once the transaction is imported, make sure a report action shows on the transaction thread of the expense showing that it was matched with a card transaction:
Screenshot 2024-05-01 at 9 12 21 PM

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

**Comment by yuwenmemon** [comment]
Removing HOLD! @shubham1206agra can you please review?

**Comment by yuwenmemon** [comment]
@shubham1206agra The tests require creating a card transaction from a personal card connected to Expensify and waiting several days for it to post, possibly. Can you just review without running through the test steps please? I can handle QA and testing once it's on staging.

**Comment by yuwenmemon** [comment]
@shubham1206agra Actually, because the test steps involve creating a personal card charge for a credit card connected to one's Expensify account, I'm going to get the review of someone internal who can do this in their dev environment more easily.

File: src/CONST.ts [hunk 1]
```diff
@@ -668,6 +668,7 @@ const CONST = {
                 MANAGER_DETACH_RECEIPT: 'MANAGERDETACHRECEIPT', // OldDot Action
                 MARKED_REIMBURSED: 'MARKEDREIMBURSED', // OldDot Action
                 MARK_REIMBURSED_FROM_INTEGRATION: 'MARKREIMBURSEDFROMINTEGRATION', // OldDot Action
+                MERGED_WITH_CASH_TRANSACTION: 'MERGEDWITHCASHTRANSACTION',
```
Comment (francoisl) [inline src/CONST.ts:671]: I don't see this action name in the backend, does it have a typo or something?

File: src/languages/es.ts [hunk 2]
```diff
@@ -3461,4 +3461,7 @@ export default {
         offline:
             'Parece que estás desconectado. Desafortunadamente, Expensify Classic no funciona sin conexión, pero New Expensify sí. Si prefieres utilizar Expensify Classic, inténtalo de nuevo cuando tengas conexión a internet.',
     },
+    systemMessage: {
+        mergedWithCashTransaction: 'encontró un recibo para esta transacción',
```
Comment (francoisl) [inline src/languages/es.ts:3465]: No period at the end like the english version?

```suggestion
        mergedWithCashTransaction: 'encontró un recibo para esta transacción.',
```

File: src/CONST.ts (hunk 1 above)
Comment (francoisl) [reply src/CONST.ts:671]: Oh nvm I was on an older branch that wasn't synced up with `main` yet.

File: src/types/onyx/OriginalMessage.ts [hunk 3]
```diff
@@ -311,6 +311,11 @@ type OriginalMessageMoved = {
     };
 };
 
+type OriginalMessageMergedWithCashTranasction = {
```
Comment (francoisl) [inline src/types/onyx/OriginalMessage.ts:314]: Typo in "transaction"

**Comment by yuwenmemon** [comment]
Updated!

**Review by francoisl — APPROVED** [review]

**Comment by francoisl** [comment]
Oh woah, the merge button was green and I didn't even realize I hadn't added the reviewer steps yet.

**Comment by yuwenmemon** [comment]
Receipt is taking forever to scan:
Screenshot 2024-05-07 at 1 58 48 PM

**Comment by yuwenmemon** [comment]
Nice this worked:
Screenshot 2024-05-07 at 4 24 35 PM

**Pull request merged**